lib/repl.js

```diff
@@ -844,10 +844,49 @@ REPLServer.prototype.createContext = function() {
   });
 
   addBuiltinLibsToObject(context);
+  previewResults.call(this);
 
   return context;
 };
 
+function previewResults() {
+  this.on('buffer', function(line) {
+
+    Interface.prototype.clearPreview.call(this);
+
+    // no need of preview for a multiline statement
+    if (this[kBufferedCommandSymbol] !== '')
+      return;
+
+    sendInspectorCommand((session) => {
+      session.post('Runtime.evaluate', {
+        expression: line.toString(),
+        generatePreview: true,
+        throwOnSideEffect: true,
+        timeout: 500,
+        executionContextId: this[kContextId]
+      }, (error, previewResult) => {
+
+        if (error) {
+          debug(`Error while generating preview ${error}`);
+        }
+
+        if (undefined !== previewResult.result.value) {
+          const value = util.inspect(previewResult.result.value);
```

Comment:
Are you sure this is what you think it is?

It currently seems to be printing Objects as Arrays of values:

```
> process.versions
{ http_parser: '2.8.0',
  node: '12.0.0-pre',
  v8: '7.0.276.38-node.11',
  uv: '1.24.0',
  zlib: '1.2.11',
  ares: '1.15.0',
  modules: '67',
  nghttp2: '1.34.0',
  napi: '3',
  openssl: '1.1.0i',
  icu: '63.1',
  unicode: '11.0',
  cldr: '34.0',
  tz: '2018e' }
> process.versions //[ '2.8.0', '12.0.0-pre', '7.0.276.38-node.11', '1.24.0', '1.2.11' ]
```